crates/bevy_pbr/src/render/mesh.rs

```diff
@@ -380,25 +386,53 @@ pub fn extract_skinned_meshes(
     commands.insert_or_spawn_batch(values);
 }
 
-fn prepare_mesh_uniforms(
+#[allow(clippy::too_many_arguments)]
+pub fn prepare_mesh_uniforms(
     mut commands: Commands,
+    mut previous_len: Local<usize>,
     render_device: Res<RenderDevice>,
     render_queue: Res<RenderQueue>,
     mut gpu_array_buffer: ResMut<GpuArrayBuffer<MeshUniform>>,
-    components: Query<(Entity, &MeshTransforms)>,
+    views: Query<(
+        &RenderPhase<Opaque3d>,
+        &RenderPhase<Transparent3d>,
+        &RenderPhase<AlphaMask3d>,
+    )>,
+    shadow_views: Query<&RenderPhase<Shadow>>,
+    meshes: Query<(Entity, &MeshTransforms)>,
 ) {
     gpu_array_buffer.clear();
 
-    let entities = components
-        .iter()
-        .map(|(entity, mesh_transforms)| {
-            (
-                entity,
-                gpu_array_buffer.push(MeshUniform::from(mesh_transforms)),
-            )
-        })
-        .collect::<Vec<_>>();
-    commands.insert_or_spawn_batch(entities);
+    let seen = FixedBitSet::new();
+    let mut indices = Vec::with_capacity(*previous_len);
+    let mut push_indices = |(mesh, mesh_uniform): (Entity, &MeshTransforms)| {
+        if !seen.contains(mesh.index() as usize) {
```

Comment:
seen is never modified here

Reply:
Looks like a mistake. seen should probably be updated with the mesh index inside the if block.

Reply:
Yeah whoops, fixed and moved the FixedBitSet to a Local like we do for visible_entities.